File: java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/elements/QuantityElement.java
package com.nedap.retail.messages.epcis.v1_1.elements;

import java.math.BigDecimal;
import java.util.Objects;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModel;

@ApiModel
public class QuantityElement {

    /**
     * A class-level identifier for the class to which the specified quantity of objects belongs.
     * For SGTIN, use URI syntax: urn:epc:id:sgtin:CompanyPrefix.ItemRefAndIndicator.SerialNumber
     * For GTIN, use URI syntax: urn:epc:id:gtin:CompanyPrefix.ItemRefAndIndicator
     */
    @JsonProperty("epc_class")
    public String epcClass;
    /**
     * A number that specifies how many or how much of the specified EPCClass is denoted by this QuantityElement.
     * Negative values are not allowed
     */
    public BigDecimal quantity;

    /**
     * Specifies a unit of measure by which the specified quantity is to be interpreted as a physical measure,
     * specifying how much of the specified EPCClass is denoted by this QuantityElement.
     */
    public String uom;

    public QuantityElement() {
    }

    public QuantityElement(final String epcClass, final BigDecimal quantity) {
        this.epcClass = epcClass;
        this.quantity = quantity;
    }

    public QuantityElement(final String epcClass, final BigDecimal quantity, final String uom) {
        this.epcClass = epcClass;
        this.quantity = quantity;
        this.uom = uom;
    }

    @Override
    public String toString() {
        return "QuantityElement: epcClass=" + epcClass + ",quantity=" + quantity + (uom != null ? ",uom=" + uom : "");
    }

    @Override
    public int hashCode() {
        int result = epcClass != null ? epcClass.hashCode() : 0;
        result = 31 * result + (quantity != null ? quantity.hashCode() : 0);
        result = 31 * result + (uom != null ? uom.hashCode() : 0);
        return result;
    }

    @Override
    public boolean equals(final Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final QuantityElement other = (QuantityElement) obj;
        if (!Objects.equals(this.epcClass, other.epcClass)) {
            return false;
        }
        if (!Objects.equals(this.quantity, other.quantity)) {
            return false;
        }
        if (!Objects.equals(this.uom, other.uom)) {
            return false;
        }
        return true;
    }
}

File: csharp/NedapRetailApiClient/Nedap.Retail.Api/Epcis.V1_1/QuantityElement.cs
using Newtonsoft.Json;
using System;

namespace Nedap.Retail.Api.Epcis.V1_1
{
    /// <summary>
    /// Quantity element
    /// </summary>
    public class QuantityElement : JsonPrintableObject
    {
        /// <summary>
        /// A class-level identifier for the class to which the specified quantity of objects belongs. EPCClass is a User Vocabulary.
        /// </summary>
        [JsonProperty("epc_class")]
        public String EpcClass { get; set; }

        /// <summary>
        /// A number that specifies how many or how much of the specified EPCClass is denoted by this QuantityElement.
        /// - If no uom is present, quantity must be an integer describing count.
        /// - If uom is present, quantity can be integer or float describing length, area, volume, or mass.
        /// Negative values are not allowed.
        /// </summary>
        [JsonProperty("quantity")]
        public float quantity { get; set; }

        /// <summary>
        /// Optional. Specifies a unit of measure by which the specified quantity is to be interpreted as a physical measure, specifying how much of the specified EPCClass is denoted by this QuantityElement.
        /// Is a 2- or 3-character code for a physical unit specified in the Common Code column of UN/CEFACT Recommendation 20.
        /// </summary>
        [JsonProperty("uom")]
        public String Uom { get; set; }


        /// <summary>
        /// Equals method for comparison of quantity elements
        /// </summary>
        /// <param name="obj">Object to compare to</param>
        /// <returns>True if quantity elements match</returns>
        public override bool Equals(object obj)
        {
            // If parameter is null return false.
            if (obj == null)
            {
                return false;
            }

            // If parameter cannot be cast to QuantityElement return false.
            QuantityElement p = obj as QuantityElement;
            if ((global::System.Object)p == null)
            {
                return false;
            }

            // Return true if the fields match:
            return Equals(p);
        }

        /// <summary>
        /// Equals method for comparison of quantity elements
        /// </summary>
        /// <param name="p">Quantity element to compare to</param>
        /// <returns>True if quantity elements match</returns>
        public bool Equals(QuantityElement p)
        {
            // If parameter is null return false:
            if ((object)p == null)
            {
                return false;
            }

            // Return true if the fields match:
            return EpcClass.Equals(EpcClass) && quantity.Equals(p.quantity) && Uom.Equals(p.Uom);
        }

        /// <summary>
        /// Override default Hashcode method
        /// </summary>
        /// <returns>Hashcode</returns>
        public override int GetHashCode()
        {
            string hashBase = "";
            if (EpcClass != null)
            {
                hashBase += EpcClass;
            }
            hashBase += quantity.ToString();
            if (Uom != null)
            {
                hashBase += Uom;
            }
            return hashBase.GetHashCode();
        }
    }
}